netmanager/src/views/components/DataDisplay/DeviceView.js

```diff
@@ -218,32 +295,226 @@ export default function DeviceView() {
             setDeviceName(ref[i].name);
             console.log('getting maintenance logs')
             console.log(logs(ref[i].name));
-            //let data = logs(ref[i].name);
-            //console.log(data);
-            //console.log(ref[i].name);
-            //let data = getMaintenanceLog(ref[i].name);
-            //console.log('setting maintenance data');
-            //console.log(data);
-            //setMaintenanceData(data);
-            //setMaintenanceData(getMaintenanceLog(ref[i].name));
             setLoaded(true);
           }
         }
     }).catch(
       console.log
     )
-    /*
-    let channelID = params.channelId
-    axios.get(constants.GET_DEVICE_UPTIME+channelID).then(({ data }) => {
-      console.log(data);
-      setNetworkUptime(data);
-    });*/
   }, []);
 
+  //Edit dialog parameters
+  const [editComponentOpen, setEditComponentOpen] = useState(false);
+  const [componentName, setComponentName] = useState('');
+  const [sensorName, setSensorName] = useState('');
+  
+  const [quantityKind, setQuantityKind] = useState([]);
+  const handleQuantityKindChange = quantity => {
+    console.log(quantity.target.value);
+    setQuantityKind(quantity.target.value);
+  } 
+
+
+  function convertQuantities(myArray){
+    console.log("Converting Quantities");
+    for (let i=0; i<myArray.length; i++){
+      //myArray[i].quantityKind = editDialogObject[myArray[i].quantityKind];
+      console.log(myArray[i].quantityKind);
+      //myArray[i].quantityKind = "Yes Please";
+    }
+    //console.log(myArray)
+    return myArray;
+  }
+  const handleSensorNameChange = name =>{
+    setSensorName(name.target.value);
+    if (name.target.value == 'Alphasense OPC-N2'){
+      setQuantityKind(["PM 1(µg/m3)", "PM 2.5(µg/m3)", "PM 10(µg/m3)"])
+    }
+    else if (name.target.value == 'pms5003'){
+      setQuantityKind(["PM 2.5(µg/m3)", "PM 10(µg/m3)"])
+    }
+    else if (name.target.value == 'DHT11'){
+      setQuantityKind(["Internal Temperature(\xB0C)", "Internal Humidity(%)"])
+    }
+    else if (name.target.value == 'Lithium Ion 18650'){
+      setQuantityKind(["Battery Voltage(V)"])
+    }
+    else if (name.target.value == 'Generic'){
+      setQuantityKind(["GPS"])
+    }
+    else if (name.target.value == 'Purple Air II'){
+      setQuantityKind(["PM 1(µg/m3)"])
+    }
+    else if (name.target.value == 'Bosch BME280'){
+      setQuantityKind(["External Temperature(\xB0C)", "External Humidity(%)"])
+    }
+    else{
+      setQuantityKind([]);
+    }
+  }
+
+  const quantityOptions = ["PM 1(µg/m3)", "PM 2.5(µg/m3)", "PM 10(µg/m3)", "External Temperature(\xB0C)", 
+  "External Temperature(\xB0F)", "External Humidity(%)", "Internal Temperature(\xB0C)", "Internal Humidity(%)", 
+  "Battery Voltage(V)", "GPS"];
+
+  const convertQuantityOptions= (myArray) => {
+    let newArray = [];
+    for (let i=0; i<myArray.length; i++){
+      if (myArray[i]=="PM 1(µg/m3)"){
```

Comment:
For those with the same measurement quantity, we could explore the possibility of using one conditional statement.

Reply:
It's a long story but it wouldn't work in this case